server.go

```diff
@@ -153,58 +154,80 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, span = tracer.Start(ctx, "ServeHTTP")
 	defer span.End()
 
-	s.PreRequest(w, r)
 	route, parameters := s.matchingRoute(r.URL.Path)
+	callbackCalled := false
 
-	if route != nil {
-		s.Logger.Printf("Handling %s\n", r.URL.Path)
-		req := multiplexer.NewRequest()
-		req.Timeout = s.ProxyTimeout
-		req.Transport = s.HttpTransport
-		req.HmacSecret = s.HmacSecret
+	if route == nil {
+		s.AroundRequest(w, r, func() {
+			s.passThrough(w, r, ctx)
+			callbackCalled = true
+		})
+	} else {
+		ctxWithPath := context.WithValue(ctx, "viewproxy-route-path", route.Path)
+		rWithPath := r.WithContext(ctxWithPath)
+		s.AroundRequest(w, rWithPath, func() {
+			s.handleRequest(w, r, route, parameters, ctx)
```

Comment:
```suggestion
			s.handleRequest(w, rWithPath, route, parameters, ctx)
```

Is there a reason we don't want to pass the route with context to `handleRequest`?

Reply:
hah! nope 😄